lib/segment/src/vector_storage/simple_vector_storage.rs

```diff
@@ -93,16 +97,34 @@ impl SimpleVectorStorage {
     /// Set deleted flag for given key. Returns previous deleted state.
     #[inline]
     fn set_deleted(&mut self, key: PointOffsetType, deleted: bool) -> bool {
-        if self.vectors.len() <= key as usize {
-            return false;
+        let was_deleted = self.deleted.replace(key as usize, deleted);
+        if !was_deleted && deleted {
```

Comment:
I propose other conditions:
```
if was_deleted != deleted {
    if deleted {
        self.deleted_count += 1;
    } else {
        self.deleted_count -= 1;
    }
}
```
I find it simpler because first `if` represents changes existing, and second `if` will specify the operation.
Also I propose to change the same condition in `grow_deleted_and_set`

Reply:
I agree that it makes it a bit more clear. Will change.